cmd/codegen/generator/go/templates/src/internal/dagger/dagger.gen.go.tmpl

```diff
@@ -4,4 +4,10 @@
 package dagger
 
 {{ template "_dagger.gen.go/defs.go.tmpl" . }}
-{{ end }}
\ No newline at end of file
+{{ end }}
+
+// Object is an interface that matches all Dagger object types
+type Object interface {
+    XXX_GraphQLType() string
+    XXX_GraphQLID(context.Context) (string, error)
+}
```

Comment:
Interesting. Should `ID()` be added, or some objects don't have an `ID()` function?

Re: Python impl., all object types extend from a base class in the library. That could be used directly as the type, unless we add the `id()` function, but even that "interface" already exists in the Python client library too.

Reply:
`ID()` can't be added unfortunately because they all have a different return type (`ContainerID`, `DirectoryID`, etc)